include/circt/Dialect/HW/HWOpInterfaces.td

```diff
@@ -36,6 +36,9 @@ def HWModuleLike : OpInterface<"HWModuleLike"> {
     "::mlir::StringAttr", "moduleNameAttr", (ins),
     /*methodBody=*/[{}],
     /*defaultImplementation=*/[{ return $_op.getNameAttr(); }]>,
+
+    InterfaceMethod<"Get the number of ports",
+    "size_t", "getNumPorts", (ins), [{}]>,
```

Comment:
FWIW this could just be `"Get the number of ports", "size_t", "getNumPorts">,`-- the rest are the defaults.  But if this is intentional to make it clear that seems fine to me :).